### ISSUE TITLE
[Bug]: mo crash when load data with 40 million rows,[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19fbb7c]

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#7795]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):6ccfbc9bfd946fafd5d60401b8c9a511bf526845
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

error message:

2023/02/02 04:40:36.481579 +0800 INFO frontend/session.go:407 session uuid: ae6bd607-a270-11ed-b491-b07b25f8b524 -> background session uuid: ae6db58f-a270-11ed-b491-b07b25f8b524
2023/02/02 04:40:36.482636 +0800 INFO frontend/util.go:515 time of Exec.Build : 1.013779ms
2023/02/02 04:40:36.482727 +0800 INFO frontend/util.go:515 time of Exec.Run : 62.045µs
2023/02/02 04:40:36.482748 +0800 INFO frontend/util.go:515 time of SendResponse 86ns
2023/02/02 04:40:36.483167 +0800 INFO rpc/handle.go:811 [precommit] handle write typ: 0, 272399-mo_mysql_compatbility_mode, 1-mo_catalog
 txn: ctx[D5C885ADEB9B4A7DBF1727C2D72D4E41][1675284036481625740-0->9223372036854775807-4294967295->9223372036854775807-4294967295][Active]: <nil>

2023/02/02 04:40:36.483472 +0800 INFO rpc/handle.go:818 [precommit] handle write end txn: ctx[D5C885ADEB9B4A7DBF1727C2D72D4E41][1675284036481625740-0->9223372036854775807-4294967295->9223372036854775807-4294967295][Active]: <nil>

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19fbb7c]

goroutine 1797 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*persistedNode).Rows(0x7f8f38192b50?)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/tables/pnode.go:81 +0x1c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*baseBlock).Rows(0xc0286d9400)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/tables/base.go:122 +0x13d
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.newAppender(...)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/tables/appender.go:33
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*ablock).MakeAppender(0xc202cbf3a0)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/tables/ablk.go:740 +0x5c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.newHandle(...)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/tables/handle.go:35
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*dataTable).GetHandle(0xc00e33fb00)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/tables/table.go:37 +0xa7
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*localSegment).prepareApplyNode(0xc16d1b5200, {0x2c9d098, 0xc147046b58})
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnimpl/localseg.go:178 +0x1f7
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*localSegment).PrepareApply(0xc16d1b5200)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnimpl/localseg.go:167 +0xd3
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnTable).PrePrepare(...)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnimpl/table.go:998

github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnTable).PrePrepare(...)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnimpl/table.go:998
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnDB).PrePrepare(0xc5e6a30480)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnimpl/txndb.go:504 +0x2d9
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnStore).PrePrepare(0xc03739d540)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnimpl/store.go:575 +0xef
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*Txn).PrePrepare(0x173fcdf2?)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnbase/txn.go:349 +0x25
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).onPrePrepare(0xc0001c9b90?, 0xc150c77ae8)
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnbase/txnmgr.go:210 +0x11f
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).dequeuePreparing(0xc0001c9b90, {0xc009ea0000, 0x1, 0x0?})
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/txn/txnbase/txnmgr.go:361 +0x42b
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start.func1()
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/logstore/sm/safeq.go:81 +0x1f1
created by github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/vm/engine/tae/logstore/sm/safeq.go:60 +0xea
[load-data.tar.gz](https://github.com/matrixorigin/matrixone/files/10564406/load-data.tar.gz)

### Steps to Reproduce

```Markdown
this test can only run on mo-srv-128

1、run a mo
2、cd /data/mo-load-data
3、./load.sh -c cases/100_columns/
```

**Comment by LeftHandCold** [comment]
fix #7822

**Comment by jiangxinmeng1** [comment]
#7822合进去了
